python/ray/data/dataset.py

```diff
@@ -2002,15 +2002,15 @@ def take_all(self, limit: int = 100000) -> List[T]:
         Returns:
             A list of all the records in the dataset.
         """
+        if limit is not None:
```

Comment:
We should follow deprecation process by first deprecating it here and in future release removing it.

Reply:
@jianoaix - we can also start with adding a `warnings.warn(""limit" argument is deprecated, and will be removed in future release", DeprecationWarning)` for 2.2, and remove in 2.3.

I don't have a strong opinion on this, both works for me, given this surface area looks small. What do other people think? @ericl and @clarkzinzow

Reply:
How about just keeping the argument and not deprecating it (only changing the default?)

Reply:
> How about just keeping the argument and not deprecating it (only changing the default?)

That also works. We can have default as None or -1. So don't enforce limit by default. This has no migration cost on user side.

Reply:
Updated to set default as None.